src/wp-includes/rest-api/endpoints/class-wp-rest-pattern-directory-controller.php

```diff
@@ -346,4 +342,40 @@ public function get_collection_params() {
 		 */
 		return apply_filters( 'rest_pattern_directory_collection_params', $query_params );
 	}
+
+	/*
+	 * Include a hash of the query args, so that different requests are stored in
+	 * separate caches.
+	 *
+	 * MD5 is chosen for its speed, low-collision rate, universal availability, and to stay
+	 * under the character limit for `_site_transient_timeout_{...}` keys.
+	 *
+	 * @link https://stackoverflow.com/questions/3665247/fastest-hash-for-non-cryptographic-uses
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array $query_args Query arguments to generate a transient key from.
+	 *
+	 * @return string Transient key.
+	 */
+	protected function get_transient_key( $query_args ) {
+
+		if ( array_key_exists( 'slug', $query_args ) ) {
+			if ( ! is_array( $query_args['slug'] ) ) {
+				$query_args['slug'] = explode( ',', (string) $query_args['slug'] );
+			}
```

Comment:
This shouldn't be necessary anymore after 5e923bc73da1322a2b64bfaf9fc090abfadfc57b.

Reply:
I left it on purpose since I didn't know the code that validates the REST request parameters very well.
I've tested it a bit, and it looks like the validation is pretty reliable, and `$query_args['slug']` can't be a string.
Fixed in 005de4aa8ae617dc0df455d433861e7a83bf4bce